### PR TITLE
Update information about project structure

### LINKED ISSUE #3246: Developer Guide doesn't fully reflect the latest structural changes
#### Description of problem:
I have read SCAP Security Guide Developer Guide https://github.com/OpenSCAP/scap-security-guide/blob/master/docs/manual/developer_guide.adoc . I think that certain parts are outdated or no longer applicable, because recently some structural changes have been done.

For example
* "Important Top Level Directory Descriptions" table doesn't describe linux_os
* 5.1.2. Product Structure/Layout is outdated 
* Directory stucture at the beginning of Section 8 Contributing with XCCDFs, OVALs and remediations doesn't contain linux_os
* formatting is broken at some places

It will be needed to review and update the text completely.

#### SCAP Security Guide Version:
current upstream master as of 3rd September 2018

#### Operating System Version:
all

#### Steps to Reproduce:

1. read SCAP Security Guide Developer Guide
2.
3.
4.

#### Actual Results:
SCAP Security Guide Developer Guide contains outdated information.

#### Expected Results:
SCAP Security Guide Developer Guide is fully updated and reflects the latest structural changes.

### PR DESCRIPTION
**Pull request by jan-cerny**
Fixes #3246

### CODE REVIEW

File: docs/manual/developer_guide.adoc [hunk 1]
```diff
@@ -469,6 +438,81 @@ directories.
...
+| Applications | `/applications` (Notice no `guide` subdirectory there!)
+| Java Runtime Environment | `/jre/guide`
+| Fuse 6 | `/fuse6/guide`
+| EAP6 | `/eap6/guide`
+| Firefox | `/firefox/guide`
+| Chromium | `/chromium/guide`
+|===
+
+The most important benchmark is *Linux OS*.
+This benchmark describes Linux Operating System in general.
```
Comment (redhatrises) [inline docs/manual/developer_guide.adoc]:
```
The most important benchmark is *Linux OS*.
This benchmark describes Linux Operating System in general.
```
should be something like
```
The *Linux OS* benchmark describes Linux Operating System in general.
```

File: docs/manual/developer_guide.adoc [hunk 2]
```diff
@@ -368,53 +368,23 @@ docker cp scap-security-guide:/home/oscap/content/build $(pwd)/container_build
...
 ===== Important Top Level Directory Descriptions
 
 |===
 |Directory |Description
 
+|`linux_os`
+| Contains security content for Linux operating systems. Contains rules, OVAL checks, Ansible tasks, Bash remeditions, ..
+
+|`applications`
+| Contains security content for applications such as OpenShift or OpenStack.
```
Comment (redhatrises) [inline docs/manual/developer_guide.adoc]: Should `Contains rules, OVAL checks, Ansible tasks, Bash remeditions, ..` be added here as well?

File: docs/manual/developer_guide.adoc (hunk 1 above)
Comment (jan-cerny) [reply docs/manual/developer_guide.adoc]: fixed

File: docs/manual/developer_guide.adoc (hunk 2 above)
Comment (jan-cerny) [reply docs/manual/developer_guide.adoc]: fixed

**Comment by redhatrises** [comment]
LGTM

**Pull request merged**